third_party/intel/lib/TritonIntelGPUToLLVM/Utility.cpp

```diff
@@ -15,10 +17,57 @@ using namespace mlir::triton;
 
 namespace mlir::LLVM::intel {
 
+static Type findShuffleType(RewriterBase &rewriter, Type valType) {
+  if (rewriter.getI8Type() == valType || rewriter.getI16Type() == valType ||
+      rewriter.getI32Type() == valType || rewriter.getI64Type() == valType ||
+      rewriter.getF16Type() == valType || rewriter.getF32Type() == valType ||
+      rewriter.getF64Type() == valType) {
+    return valType;
+  }
+  if (rewriter.getBF16Type() == valType) {
+    return rewriter.getI16Type();
+  }
+
+  unsigned bitWidth = valType.getIntOrFloatBitWidth();
+  if (bitWidth < 8) {
+    return rewriter.getI8Type();
+  }
+  return {};
+}
+
 static Value shuffleCommon(Location loc, RewriterBase &rewriter, Value val,
-                           Value i, TritonGEN::ShflKind mode) {
-  Type type = val.getType();
-  return rewriter.create<TritonGEN::SubGroupShuffleOp>(loc, type, val, i, mode);
+                           Value i, mlir::gpu::ShuffleMode mode) {
+  Type valType = val.getType();
+  Type shuffleType = findShuffleType(rewriter, valType);
+
+  unsigned bitWidth = valType.getIntOrFloatBitWidth();
+  if (shuffleType != valType) {
+    assert(shuffleType.isInteger());
+    if (!valType.isInteger()) {
+      val = bitcast(val, int_ty(bitWidth));
+    }
+    if (bitWidth < shuffleType.getIntOrFloatBitWidth()) {
+      val = zext(shuffleType, val);
+    }
+  }
+
+  int width = TritonGEN::getSubgroupSize(i.getDefiningOp());
+  Value widthConstant =
+      rewriter.create<arith::ConstantIntOp>(loc, width, 32).getResult();
```

Comment:
```suggestion
  Value widthConstant = int_32(width);
```

Or at least generate `LLVM::ConstantOp`

Reply:
I'm not sure about you suggestion because I couldn't see a macro/function call `int_32`, but I've done the ConstantOp

Reply:
I think you meant `i32_val`